app/models/search/tag_indexer.rb

```diff
@@ -52,8 +52,28 @@ def self.settings
   def document(object)
     object.as_json(
       root: false,
-      only: [:id, :name, :merger_id, :canonical, :created_at]
-    ).merge(tag_type: object.type)
+      only: [:id, :name, :sortable_name, :merger_id, :canonical, :created_at]
+    ).merge(
+      tag_type: object.type,
+      uses: object.taggings_count_cache
+    ).merge(parent_data(object))
+  end
+
+  # Index parent data for tag wrangling searches
+  def parent_data(tag)
+    data = {}
+    %w(Media Fandom Character).each do |parent_type|
+      if tag.parent_types.include?(parent_type)
+        key = "#{parent_type.downcase}_ids"
+        ids = tag.parents.by_type(parent_type).pluck(:id)
+        # add a dummy value so we can find the unwrangled tags more easily
+        # since you can't search for an empty array
+        data[key] = ids.empty? ? [0] : ids
```

Comment:
FYI, it looks like there is a query that lets you search for an empty array:
https://www.elastic.co/guide/en/elasticsearch/reference/6.4/query-dsl-exists-query.html

Reply:
Kibana tells me this field is a number, so I _think_ we'd have to set it to `0` anyway: https://www.elastic.co/guide/en/elasticsearch/reference/current/number.html

It seems like this method is ultimately a less verbose way of doing the same thing

Reply:
I meant the exists query, not the null_value mapping. I suspect something like this should work, though I haven't finished [running all of the tests](https://travis-ci.org/tickinginstant/otwarchive/builds/459989946):

https://github.com/tickinginstant/otwarchive/commit/499256ade703fce97aff6102319074111349802a

Reply:
I prefer using 0 for empty (basically ad-hoc null_value mapping), since it means not having to flip the logic in my head for an exclusion filter. If the code looks kinda mysterious, we can do something like:

```rb
TagQuery.new(options.merge(
  no_fandom: true,
))
```
Then in the query class:
```rb
def no_fandom_filter
  terms_filter(:fandom_ids, [0]) if options[:no_fandom]
end
```
then add `no_fandom_filter` to `filters` instead of `exclusion_filters`. Add some comments in (plus a test in tag_query_spec.rb) and it should be pretty readable.

Reply:
I'd also call it `unwrangled_filter` rather than no fandom, to avoid any confusion between unwrangled tags and no fandom tags (which actually have a fandom called "No Fandom").

Reply:
I disagree about using 0 for empty. The exists query accomplishes the same thing without having to modify the data (which means that there is less danger of running into future issues where the 0 is accidentally treated as real data -- e.g. in aggregations, or classes like PseudDecorator -- and also means that related techniques can be used in other query classes without having to reindex everything to add in the null value).

If it absolutely has to be in the list of positive filters, it can be wrapped in another bool query as a `must_not`. But I think it's cleaner to just have it be an exclusion filter -- probably with a better name, since that seems to be an issue.